patches/chrome-browser-themes-theme_service.cc.patch

```diff
@@ -0,0 +1,25 @@
+diff --git a/chrome/browser/themes/theme_service.cc b/chrome/browser/themes/theme_service.cc
+index 655be29bffb9c3c6abc063c255fac06aaa69951d..5a0366963f8c39cc32f65ce55533d38de8e38b74 100644
+--- a/chrome/browser/themes/theme_service.cc
++++ b/chrome/browser/themes/theme_service.cc
+@@ -64,6 +64,10 @@
+ #include "chrome/browser/supervised_user/supervised_user_theme.h"
+ #endif
+ 
++#if defined(BRAVE_CHROMIUM_BUILD)
++#include "brave/browser/themes/theme_properties.h"
++#endif
++
+ using base::UserMetricsAction;
+ using content::BrowserThread;
+ using extensions::Extension;
+@@ -157,6 +161,9 @@ gfx::ImageSkia* ThemeService::BrowserThemeProvider::GetImageSkiaNamed(
+ }
+ 
+ SkColor ThemeService::BrowserThemeProvider::GetColor(int id) const {
++#if defined(BRAVE_CHROMIUM_BUILD)
++  MAYBE_OVERRIDE_DEFAULT_COLOR_FOR_BRAVE(id, incognito_, theme_service_.profile())
```

Comment:
why aren't we just using a theme service with the color we want?

Reply:
Clients can get color value from `ui::ThemeProvider` and BrowserThemeProvider is the implementation.

Reply:
I don't understand. The ThemeService comes from ThemeServiceFactory and we can change that by creating our own and doing a preprocessor replace in ThemeServiceFactory

Reply:
`#define ThemeService BraveThemeService`, `#define ThemeServiceWin BraveThemeServiceWin`, etc... no patch required

Reply:
what we're doing right now seems doable with a built-in theme extension

Reply:
oh, actually I think those are just leftover, they don't seem to hooked up to anything anymore

Reply:
@bridiver We decided in discussion with @bbondy not to use multiple theme extensions in order to:
1. support varying layout properties and colors which aren't themeable via an extension
2. avoid flash of unthemed content

If we can avoid 2. then I suppose we could use a theme in addition to modifying other properties, but seems simpler to manage having it all versioned in the same place.

Reply:
either way we can create a custom theme without using extensions

Reply:
probably override UseDefaultTheme/SetCustomDefaultTheme

Reply:
Oh! I think It's worth to try CustomThemeSupplier.

Reply:
@bbondy , Have you considered using `CustomThemeSupplier` when you implement https://github.com/brave/brave-core/pull/328/?

Reply:
@bridiver this is going back to https://github.com/brave/brave-core/pull/328 then, but as long as we can still:
- override properties inside `NotOverwritableByUserThemeProperty`
- fallback values we're not re-defining to the existing chromium default values
- allow user theme extensions via themepack

...then maybe it is preferrable for us to make our own ThemeProvider considering it will allow overriding more than just colors. Also we will at some point need to know about tor, and right now the ThemeProvider just gets instantiated with incognito (i.e. we'll have 6 built-in theme variations - Light{Normal,Private,Tor} and Dark{Normal,Private,Tor} with some amount of fallback.

Reply:
actually I had Tor in mind because we can change the ThemeSupplier based on the Profile

Reply:
There's also
```
// static
const ui::ThemeProvider& ThemeService::GetThemeProviderForProfile(
    Profile* profile) 
```

Reply:
I think `CustomThemeSupplier` is not an option for us if we will allow user theme extension install.
Only one theme supplier is used at once.

Reply:
If we want to override **default** colors,
calling `MAYBE_OVERRIDE_DEFAULT_COLOR_FOR_BRAVE(id, incognito, profile())`
before `GetDefaultColor(id, incognito)` like below is not bad.
```
SkColor ThemeService::GetColor(int id,
                               bool incognito,
                               bool* has_custom_color) const {
  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);

  if (has_custom_color)
    *has_custom_color = false;

  // The incognito NTP always uses the default background color, unless there is
  // a custom NTP background image. See also https://crbug.com/21798#c114.
  if (id == ThemeProperties::COLOR_NTP_BACKGROUND && incognito &&
      !HasCustomImage(IDR_THEME_NTP_BACKGROUND)) {
    return ThemeProperties::GetDefaultColor(id, incognito);
  }

  SkColor color;
  const int theme_supplier_id = incognito ? GetIncognitoId(id) : id;
  if (theme_supplier_ && theme_supplier_->GetColor(theme_supplier_id, &color)) {
    if (has_custom_color)
      *has_custom_color = true;
    return color;
  }

#if defined(BRAVE_CHROMIUM_BUILD)
  MAYBE_OVERRIDE_DEFAULT_COLOR_FOR_BRAVE(id, incognito, profile())
#endif

  return GetDefaultColor(id, incognito);
}
```

With this, user installed theme is applied(if exist) and we can override `NotOverwritableByUserThemeProperty` properties.

Reply:
BraveThemeService does not need patching

Reply:
and we don't even necessarily need our own CustomThemeSupplier if we're overriding ThemeService